#! /bin/sh -e
# 02_devmapper1.02.dpatch by Bastian Blank <waldi@debian.org>
#
# All lines beginning with `## DP:' are a description of the patch.
## DP: fix lib/libdevmapper.c for devmapper 1.02

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1;;
esac

exit 0

@DPATCH@
--- cryptsetup-1.0.1.orig/lib/libdevmapper.c
+++ cryptsetup-1.0.1/lib/libdevmapper.c
@@ -226,6 +226,11 @@
 	if (!dm_task_get_info(dmt, &dmi))
 		goto out;
 
+	if (!dmi.exists) {
+		r = -ENODEV;
+		goto out;
+	}
+
 	next = dm_get_next_target(dmt, next, &start, &length,
 	                          &target_type, &params);
 	if (!target_type || strcmp(target_type, CRYPT_TARGET) != 0 ||
